src/ansiblelint/utils.py

```diff
@@ -103,6 +103,8 @@ def path_dwim(basedir: str, given: str) -> str:
 def ansible_template(
     basedir: str, varname: Any, templatevars: Any, **kwargs: Any
 ) -> Any:
+    if os.path.basename(basedir) == 'tasks':
+        basedir = os.path.dirname(basedir)
```

Comment:
Can you please explain why this fixes the issue? I am not too intimately familiar with this part of the codebase so that explanation would help. Things like how the `block` influences the `basedir` value, ...

Reply:
I added some comments about this fix and why blocks are needed. They are actually two issues.